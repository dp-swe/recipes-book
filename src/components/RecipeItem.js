import React from 'react';
import styles from './styles/recipeItem.module.scss';
import texts from './texts';

const getIngredientsList = (ingredients) => (
    ingredients
    .split(",")
    .map((ingredient, index) => 
        <li key={index+ingredient} className={styles.recipe__ingredient}>
            {ingredient.trim()}
        </li>
    )
);

const checkLactose = (ingredients) => ingredients.match(/milk|cheese/i);

const RecipeItem = ({item}) => (
    <>
        <a href={item.href} className={styles.recipe__a}>
            <div className={styles.recipe__image_wrapper}>
                {checkLactose(item.ingredients) &&
                    <span className={styles.recipe__label}>{texts.LABEL_LACTOSE}</span>
                }
                <img className={styles.recipe__image} alt={"Recipe"} src={item.thumbnail}/>
            </div>
            <div className={styles.recipe__title}>
                {item.title}   
            </div>     
            <button className={styles.recipe__button}>
                    {texts.BUTTON_FAVORITE}
            </button>
            <ul className={styles.recipe__ingredients}>
                {getIngredientsList(item.ingredients)}
            </ul>
        </a>
    </>
)

export default RecipeItem;