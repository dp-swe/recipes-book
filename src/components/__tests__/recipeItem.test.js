import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import RecipeItem from '../RecipeItem';

describe('RecipeItem Component', () => {
    let item;
    let wrapper;

    beforeEach(() => {
        item = {
                title: "Vegetable-Pasta Oven Omelet",
                href: "http://find.myrecipes.com/recipes/recipefinder.dyn?action=displayRecipe&recipe_id=520763",
                ingredients: "tomato, milk, red pepper",
                thumbnail: "http://img.recipepuppy.com/560556.jpg"
                };
        wrapper = shallow(<RecipeItem item={item}/>);
    })

    it('should render without broken', () => {
        expect(wrapper.exists()).toBe(true);
    })

    it('should render correctly', () => {
        const tree = renderer.create(<RecipeItem item={item}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
 
    it('should contain a list of ingredients', () => {
        expect(wrapper.find("li")).toHaveLength(3);
        expect(wrapper.find("li").get(0).props.children).toEqual('tomato');
        expect(wrapper.find("li").get(1).props.children).toEqual('milk');
        expect(wrapper.find("li").get(2).props.children).toEqual('red pepper');
    })

    it('should contain lactose label if ingredients contain milk or cheese', () => {
        expect(wrapper.find(".recipe__label")).toHaveLength(1);
    })

    it('should not contain a list of ingredients', () => {
        item = {
                title: "Vegetable-Pasta Oven Omelet",
                href: "http://find.myrecipes.com/recipes/recipefinder.dyn?action=displayRecipe&recipe_id=520763",
                ingredients: "",
                thumbnail: "http://img.recipepuppy.com/560556.jpg"
            };
        wrapper = shallow(<RecipeItem item={item}/>);

        expect(wrapper.find("li")).toEqual({});
    })

    it('should not contain lactose label', () => {
        item = {
            title: "Vegetable-Pasta Oven Omelet",
            href: "http://find.myrecipes.com/recipes/recipefinder.dyn?action=displayRecipe&recipe_id=520763",
            ingredients: "tomato, cheese, red pepper",
            thumbnail: "http://img.recipepuppy.com/560556.jpg"
            };

        wrapper = shallow(<RecipeItem item={item}/>);

        expect(wrapper).not.toContain(".recipe__label");
    })

})