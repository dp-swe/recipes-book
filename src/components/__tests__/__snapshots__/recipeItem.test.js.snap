// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeItem Component should render correctly 1`] = `
<a
  className="recipe__a"
  href="http://find.myrecipes.com/recipes/recipefinder.dyn?action=displayRecipe&recipe_id=520763"
>
  <div
    className="recipe__image_wrapper"
  >
    <span
      className="recipe__label"
    >
      Has Lactose
    </span>
    <img
      alt="Recipe"
      className="recipe__image"
      src="http://img.recipepuppy.com/560556.jpg"
    />
  </div>
  <div
    className="recipe__title"
  >
    Vegetable-Pasta Oven Omelet
  </div>
  <button
    className="recipe__button"
  >
    Make Favorite
  </button>
  <ul
    className="recipe__ingredients"
  >
    <li
      className="recipe__ingredient"
    >
      tomato
    </li>
    <li
      className="recipe__ingredient"
    >
      milk
    </li>
    <li
      className="recipe__ingredient"
    >
      red pepper
    </li>
  </ul>
</a>
`;
